internal/p2p/network_messenger.go

```diff
@@ -0,0 +1,155 @@
+package p2p
+
+import (
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/libsv/go-p2p/wire"
+)
+
+type NetworkMessanger struct {
+	pm *PeerManager
+}
+
+func NewHerald(m *PeerManager) *NetworkMessanger {
+	return &NetworkMessanger{pm: m}
+}
+
+func (h *NetworkMessanger) CountConnectedPeers() uint {
+	return h.pm.CountConnectedPeers()
+}
+
+// AnnounceTransactions will send an INV messages to the provided peers or to selected peers if peers is nil.
+// It will return the peers that the transaction was actually announced to.
+func (h *NetworkMessanger) AnnounceTransactions(txHashes []*chainhash.Hash, peers []PeerI) []PeerI {
```

Comment:
Nit-pick: why `h`? 😭 Is it from the previous 'NetworkHerald`? Maybe change to `m` or `nm`?